lib/charms/oathkeeper/v0/auth_proxy.py

```diff
@@ -337,6 +342,33 @@ def _on_relation_departed_event(self, event: RelationDepartedEvent) -> None:
         """Notify Oathkeeper that the relation has departed."""
         self.on.config_removed.emit(event.relation.id)
 
+    def get_headers(self) -> Optional[List[str]]:
+        """Returns the list of headers from all relations."""
+        if len(self.model.relations) == 0:
+            return None
+
+        headers = list()
+        for relation in self._charm.model.relations[self._relation_name]:
+            if not relation.data[relation.app]:
+                return None
+            for header in json.loads(relation.data[relation.app]["headers"]):
+                # Avoid duplicates
+                if header not in headers:
+                    headers.append(header)
```

Comment:
You can make headers a set to avoid having to check if the header still exists. You can cast to list on return.